plugin.php

```diff
@@ -84,94 +83,98 @@ function create_initial_json_routes() {
 	/*
 	 * Posts
 	 */
-	$controller = new WP_JSON_Posts_Controller;
-	register_json_route( 'wp', '/posts', array(
-		array(
-			'methods'         => WP_JSON_Server::READABLE,
-			'callback'        => array( $controller, 'get_items' ),
-			'args'            => array(
-				'context'          => array(
-					'default'      => 'view',
+	foreach( get_post_types( array( 'public' => true ) ) as $post_type ) {
+		$controller = new WP_JSON_Posts_Controller;
+		$controller->post_type = $post_type;
+		$base = $post_type . 's';
```

Comment:
Hmm, not such a fan of doing _anything_ automatic here - if you register a post type through `register_post_type`, and you never explicitly declare the json route url, it doesn't really follow that the path to your CPT is something _you never entered_. Auto pluralisation is just a bit too magic for my liking.